cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/LocalArtifactLoaderStage.java

```diff
@@ -87,8 +87,8 @@ public void process(AppDeploymentInfo deploymentInfo)
     String appVersion = deploymentInfo.getApplicationVersion();
     String configString = deploymentInfo.getConfigString();
 
-    NamespacedImpersonator classLoaderImpersonator =
-      new NamespacedImpersonator(artifactId.getParent(), impersonator);
+    EntityImpersonator classLoaderImpersonator =
+      new EntityImpersonator(artifactId.getParent(), impersonator);
```

Comment:
I tried to do that but owner store did not support EntityType.ARTIFACT at that point. Now that is almost ready, I will add artifact there.